### PR TITLE
Pure V6 scale-out sample

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Shows V6 scale-out options:
 * competing consumers
 * sender-side distribution

### CODE REVIEW

File: samples/scaleout/senderside/Version_6/AwareClient/AwareClient.csproj [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    <DebugType>pdbonly</DebugType>
+    <Optimize>true</Optimize>
+    <OutputPath>bin\Release\</OutputPath>
+    <DefineConstants>TRACE</DefineConstants>
+    <ErrorReport>prompt</ErrorReport>
+    <WarningLevel>4</WarningLevel>
+  </PropertyGroup>
+  <ItemGroup>
+    <Reference Include="NServiceBus.Core, Version=6.0.0.0, Culture=neutral, PublicKeyToken=9fc386479f8a226c, processorArchitecture=MSIL">
+      <HintPath>..\..\..\..\..\packages\NServiceBus.6.0.0-pullrequest3374\lib\net452\NServiceBus.Core.dll</HintPath>
```
Comment (SimonCropp) [inline samples/scaleout/senderside/Version_6/AwareClient/AwareClient.csproj]: Just an fyi this cant be merged with a pr nuget ref

File: samples/scaleout/senderside/Version_6/AwareClient/AwareClient.csproj [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      <Name>Messages</Name>
+    </ProjectReference>
+  </ItemGroup>
+  <ItemGroup>
+    <Content Include="routes.xml">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+  </ItemGroup>
+  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+  <!-- To modify your build process, add your task inside one of the targets below and uncomment it.
```
Comment (SimonCropp) [inline samples/scaleout/senderside/Version_6/AwareClient/AwareClient.csproj]: Delete the comment

File: samples/scaleout/senderside/Version_6/AwareClient/Program.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+using Messages;
+using NServiceBus;
+
+namespace AwareClient
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            BusConfiguration busConfig = new BusConfiguration();
```
Comment (SimonCropp) [inline samples/scaleout/senderside/Version_6/AwareClient/Program.cs]: Can u rename to be consistent on busconfiguration. Will make it easier when we "endpointconfig"

File: samples/scaleout/senderside/Version_6/AwareClient/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
```
Comment (SimonCropp) [inline samples/scaleout/senderside/Version_6/AwareClient/Properties/AssemblyInfo.cs]: Delete this

File: samples/scaleout/senderside/Version_6/Server1/Program.cs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using Messages;
+using NServiceBus;
+
+namespace Server1
+{
+    class Program
+    {
+        static void Main(string[] args)
```
Comment (SimonCropp) [inline samples/scaleout/senderside/Version_6/Server1/Program.cs]: Remove the redundant args

File: samples/scaleout/senderside/Version_6/Server1/Program.cs [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+            #region Server-Set-InstanceId
+            busConfig.EndpointInstanceId(() => ConfigurationManager.AppSettings["InstanceId"]);
+            #endregion
+
+            busConfig.UsePersistence<InMemoryPersistence>();
+            busConfig.SendFailedMessagesTo("error");
+            Run(busConfig).GetAwaiter().GetResult();
+        }
+
+        private static async Task Run(BusConfiguration busConfig)
```
Comment (SimonCropp) [inline samples/scaleout/senderside/Version_6/Server1/Program.cs]: Redundant private

File: samples/scaleout/senderside/Version_6/Server1/Program.cs [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            IEndpointInstance endpoint = await Endpoint.Start(busConfig);
+            Console.WriteLine("Press <enter> to exit.");
+            Console.ReadLine();
+            await endpoint.Stop();
+        }
+    }
+
+    #region Server-Handler
+    public class DoSomethingHandler : IHandleMessages<DoSomething>
```
Comment (SimonCropp) [inline samples/scaleout/senderside/Version_6/Server1/Program.cs]: Split to a diff file

File: samples/scaleout/senderside/sample.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+These two projects are identical and they are there to show the scale out in action from within the Visual Studio. In a real-world scenario there would be a single Server project that is deployed to multiple paths and/or machines.
+
+#### Instance ID
+
+Following snippet instructs NServiceBus to get the instance ID from the `appSettings` configuration section.
+
+snippet:Server-Set-InstanceId
+snippet:Server-InstanceId
+
+WARNING: You should never hardcode instance IDs because they are mostly operations concern and operations people should be able to modify them without the need to recompile the source code.
```
Comment (SimonCropp) [inline samples/scaleout/senderside/sample.md]: Remove you

File: samples/scaleout/senderside/sample.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+snippet:File-Based-Routing
+
+snippet:Physical-Routes
+
+## Running the code
+
+Start the solution with all the console applications (`Server1`, `Server2`, `UnawareClient`, `AwareClient`) as startup applications.
+
+### Competing consumers
+
```
Comment (SimonCropp) [inline samples/scaleout/senderside/sample.md]: Remove you

File: samples/scaleout/senderside/sample.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+Go to the `UnawareClient` console an press enter a few times. You should see `Message received.` messages being printed out on `Server1` and `Server2` consoles. Notice that these messages seem to be routed randomly to either of the servers and there are cases when a number of consecutive messages end up in a single server (consequence of randomness). This is because the unaware client uses the queue shared by both servers and the message is processed by the first one who gets it from the underlying queue.
+
+### Round robin
+
+Now go to the `AwareClient` console and, again, press enter a few times. You should notice that `Message received.` messages show up alternately in `Server1` and `Server2` windows. This is because the client is in charge of the distribution and sends his messages to instance-specific queues.
```
Comment (SimonCropp) [inline samples/scaleout/senderside/sample.md]: Remove you

Comment (SimonCropp) [inline samples/scaleout/senderside/sample.md]: Remove his

**Comment by SimonCropp** [comment]
i fixed my comments

**Comment by SimonCropp** [comment]
@SzymonPobiega as suspected this needs a rebase. i might get to it today

**Comment by SimonCropp** [comment]
merged manually here https://github.com/Particular/docs.particular.net/commit/1f9d4cbdc3ecf42d43f0a32ba37eee8dc2a8950f

@SzymonPobiega feel free to QA that commit if u have time

**Comment by SzymonPobiega** [comment]
:+1:

**Pull request closed without merging**